src/pohjavirta/server.clj

```diff
@@ -34,8 +34,13 @@
                    :else (reify HttpHandler
                            (handleRequest [_ exchange]
                              (let [request (request/create exchange)
-                                   response (handler request)]
-                               (response/send-response response exchange)))))]
+                                   handler (fn [req]
+                                             (-> req
+                                                 (handler)
+                                                 (response/send-response exchange)))]
+                               (if (fn? (:body request))
+                                 ((:body request) handler request)
```

Comment:
Looks good, but is not optimal in perf I believe: `:body` is looked up twice, and as it's backed by the slow Potemkin lazy-map, we lose 2*70ns here. I would suggest:

* adding the `:body` to `pohjavirta.request.Request` for fast access (few nanos)
* maybe add a Protocol like `pohjavirta.request.HandleWithBody` that has `(handle-with-body [_ handler])` function that is invoked in `pohjavirta.server` and would do things behind the scenes. To remove the `if` completely, there could be `AsyncPartialRequest` and `SyncPartialRequest` that have their own implementation of `HandleWithBody`, as we know how to parse the body already when server is created (with `:dispatch` or not).